### PR TITLE
BPE Detokenize

### PR DESCRIPTION
**Pull request by pforderique**
Implements the detokenize method for BytePairEncoding.

Note:
Kera’s detokenize() currently uses Python’s byte strings (b’my text here’) so their implementation is a bit different than what I have here. I’ve added a test case for now that works on both implementations, highlighting that `detokenize(tokenize(input))` should equal the original `input`, but there might be something I’m missing due to the omitted part of the implementation. This can be pushed off to the clean up PR in the future, but should be okay for now.

### CODE REVIEW

**Comment by mattsoulanille** [comment]
LGTM once #7780 is merged.

**Review by mattsoulanille — APPROVED** [review]

**Review by Linchenn — APPROVED** [review]
LGTM!

**Pull request merged**
